dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/icon/CustomIconRequest.java

```diff
@@ -28,20 +28,27 @@
 package org.hisp.dhis.webapi.controller.icon;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Set;
 import lombok.AllArgsConstructor;
+import lombok.Builder;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 
-@NoArgsConstructor
-@AllArgsConstructor
+@Builder
 @Getter
-class IconDto {
+@AllArgsConstructor
+@NoArgsConstructor
+class CustomIconRequest {
 
   @JsonProperty private String key;
 
+  @JsonProperty private String code;
+
   @JsonProperty private String description;
 
-  @JsonProperty private String[] keywords;
+  @JsonProperty private Set<String> keywords;
+
+  @JsonProperty private String fileResourceId;
 
-  @JsonProperty private String fileResourceUid;
+  @Builder.Default @JsonProperty private Boolean custom = true;
```

Comment:
Is this only used to create/update custom icons? if so we would not need this field

Reply:
This is so we can distinguish user provided (custom) icons from bundled icons later assuming they all are in the same table(s)

Reply:
Sure but I this class is used in the controller for POST addIcon and PUT updateIcon. I assume a default icon cannot be created or updated, right? If that is the case it does not make sense to add a flag. It would then also be clearer if this is called CustomIconRequest.